src/Nancy.Swagger/SwaggerConfig.cs

```diff
@@ -52,16 +55,38 @@ static SwaggerConfig()
         /// And so on.</remarks>
         public static string DefaultModelIdConvention(Type type)
         {
-            var id = type.Name;
-            var fullName = type.FullName;
-            var idx = fullName.LastIndexOf('.');
-            while (_resolvedModelIds.GetOrAdd(id, type) != type)
+            //var id = type.Name;
+            //var fullName = type.FullName;
+            //var idx = fullName.LastIndexOf('.');
+            //while (_resolvedModelIds.GetOrAdd(id, type) != type)
+            //{
+            //    idx = fullName.LastIndexOf('.', idx - 1);
+            //    id = idx <= 0 ? fullName : fullName.Substring(idx).Replace(".", "");
+            //}
+
+            //return id;
```

Comment:
Actually I think I'm losing some important functionality there -- with adding pieces of the fully qualified name in case of collisions. I'm adding this back in, but underneath the check for generics and containers